### PR TITLE
MODFEE-81 create check-transfer endpoint, add tests

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Approach
- create check-transfer endpoint;
- add validation;
- create tests;

Resolves: [MODFEE-81](https://issues.folio.org/browse/MODFEE-81)

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -218,6 +218,15 @@
...
           ]
         },
+        {
+          "methods":[
+            "POST"
+          ],
+          "pathPattern":"/accounts/{id}/check-transfer",
+          "permissionsRequired":[
+            "accounts.check-transfer.post"
+          ]
```
Comment (alexanderkurash) [inline descriptors/ModuleDescriptor-template.json:228]: Permission set?

File: src/test/java/org/folio/rest/impl/AccountsActionChecksAPITests.java [hunk 2]
```diff
@@ -30,6 +31,14 @@ public void checkWaiveAmountShouldBeAllowed() {
     actionCheckAmountShouldBeAllowed(accountsCheckWaiveClient, accountToPost);
   }
 
+  @Test
+  public void checkTransferAmountShouldBeAllowed() {
+    Account accountToPost = postAccount();
```
Comment (alexanderkurash) [inline src/test/java/org/folio/rest/impl/AccountsActionChecksAPITests.java:36]: What else do we need from AccountsAPITest?

File: src/test/java/org/folio/rest/impl/AccountsActionChecksAPITests.java [hunk 3]
```diff
@@ -44,6 +53,14 @@ public void checkWaiveAmountShouldNotBeAllowedWithExceededAmount() {
     actionCheckAmountShouldNotBeAllowedWithExceededAmount(accountsCheckWaiveClient);
   }
 
+  @Test
+  public void checkTransferAmountShouldNotBeAllowedWithExceededAmount() {
+    Account accountToPost = postAccount();
+    ResourceClient accountsCheckTransferClient =
+      accountsCheckTransferClient(accountToPost.getId());
```
Comment (alexanderkurash) [inline src/test/java/org/folio/rest/impl/AccountsActionChecksAPITests.java:60]: Can we move this to @Before or @BeforeClass?

File: descriptors/ModuleDescriptor-template.json (hunk 1 above)
Comment (roman-barannyk) [reply descriptors/ModuleDescriptor-template.json:228]: added

File: src/test/java/org/folio/rest/impl/AccountsActionChecksAPITests.java (hunk 2 above)
Comment (roman-barannyk) [reply src/test/java/org/folio/rest/impl/AccountsActionChecksAPITests.java]: updated

File: src/test/java/org/folio/rest/impl/AccountsActionChecksAPITests.java (hunk 3 above)
Comment (roman-barannyk) [reply src/test/java/org/folio/rest/impl/AccountsActionChecksAPITests.java]: moved

File: descriptors/ModuleDescriptor-template.json [hunk 4]
```diff
@@ -814,6 +814,26 @@
       "displayName":"feefineactions item put",
       "description":"Edit an feefineaction record"
     },
+    {
```
Comment (alexanderkurash) [inline descriptors/ModuleDescriptor-template.json:817]: Not the best placement

File: src/test/java/org/folio/test/support/ApiTests.java [hunk 5]
```diff
@@ -129,4 +132,21 @@ private static String generateOkapiToken() {
...
+      .withLoanId(randomId())
+      .withMaterialTypeId(randomId())
+      .withFeeFineId(randomId())
+      .withFeeFineType("book lost")
+      .withFeeFineOwner("owner")
+      .withAmount(9.00)
+      .withRemaining(4.55)
+      .withPaymentStatus(new PaymentStatus().withName("Outstanding"))
+      .withStatus(new Status().withName("Open"));
+  }
```
Comment (alexanderkurash) [inline src/test/java/org/folio/test/support/ApiTests.java:151]: Move to a helper class (EntityBuilder)?

Comment (roman-barannyk) [reply src/test/java/org/folio/test/support/ApiTests.java]: moved

File: descriptors/ModuleDescriptor-template.json (hunk 4 above)
Comment (roman-barannyk) [reply descriptors/ModuleDescriptor-template.json]: moved

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**